docs/manual/cluster-admin/how-to-customize-alerts.md

```diff
@@ -0,0 +1,152 @@
+  email-notification: # email-notification will only be enabled when this field is not empty
+    receiver: your_addr@example.com
+    smtp_url: smtp.office365.com:587
+    smtp_from: alert_sender@example.com
+    smtp_auth_username: alert_sender@example.com
+    smtp_auth_password: password_for_alert_sender
+  webhook-actions: # webhook-actions will only be enabled when  this field is not empty
+    port: 9095 # optional, do not change this if you do not want to change the port alert-handler is listening on
+    bearer_token: 'your_application_token_for_rest_server'
+```
+
+### How to Match Alerts and Actions
+
+The matching rules of alerts and actions are defined in [alert-configmap.yaml.template](https://github.com/microsoft/pai/blob/master/src/alert-manager/deploy/alert-configmap.yaml.template):
+
+``` yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: alertmanager
+data:
+  config.yml: |-
+{% set email_notification = cluster_cfg["alert-manager"]["email-notification"] %}
+{% set webhook_actions = cluster_cfg["alert-manager"]["webhook-actions"] %}
+    global:
+      resolve_timeout: 5m
+{% if cluster_cfg["alert-manager"]["email-notification-configured"] %}
+      smtp_smarthost: {{ email_notification["smtp_url"] }}
+      smtp_from: {{ email_notification["smtp_from"] }}
+      smtp_auth_username: {{ email_notification["smtp_auth_username"] }}
+      smtp_auth_password: {{ email_notification["smtp_auth_password"] }}
+{% endif %}
+    
+    templates:
+    - "/etc/alertmanager/template/*.tmpl"
+
+    route:
+      receiver: pai-email
+      group_wait: 30s
+      group_interval: 5m
+      repeat_interval: {{ cluster_cfg["alert-manager"]["repeat-interval"] }}
+      group_by: [alertname, alertstate]
+      routes:
+      - receiver: pai-email-and-stop-job
+        match: 
+          alertname: PAIJobGpuPercentLowerThan0_3For1h
+    
+    receivers:
+    - name: "pai-email"
+{% if cluster_cfg["alert-manager"]["email-notification-configured"] %}
+      email_configs:
+        - to: {{ email_notification["receiver"] }}
+          send_resolved: true
+          html: '{{ "{{" }} template "email.pai.html" . {{ "}}" }}'
+          headers:
+            subject: '{{ cluster_cfg["cluster"]["common"]["cluster-id"] }}: {{ "{{" }} template "__subject" . {{ "}}" }}'
+{% endif %}
+    - name: "pai-email-and-stop-job"
+{% if cluster_cfg["alert-manager"]["email-notification-configured"] %}
+      email_configs:
+        - to: {{ email_notification["receiver"] }}
+          send_resolved: true
+          html: '{{ "{{" }} template "email.pai.html" . {{ "}}" }}'
+          headers:
+            subject: '{{ cluster_cfg["cluster"]["common"]["cluster-id"] }}: {{ "{{" }} template "__subject" . {{ "}}" }}'
+{% endif %}
+{% if cluster_cfg["alert-manager"]["webhook-actions-configured"] %}
+      webhook_configs:
+        - url: 'http://localhost:{{ webhook_actions["port"] }}/alert-handler/stop-job'
```

Comment:
Seems we only can define `stop-job` webhook?  What if we want to add more webhooks, such as uncordon node

Reply:
> Seems we only can define `stop-job` webhook? What if we want to add more webhooks, such as uncordon node

We'll need to add an API like `alert-handler/uncordon-node` in alerthandler.